In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("train.csv.zip")

In [3]:
df

ID_code  target    var_0   var_1    var_2   var_3    var_4  \
0            train_0       0   8.9255 -6.7863  11.9081  5.0930  11.4607   
1            train_1       0  11.5006 -4.1473  13.8588  5.3890  12.3622   
2            train_2       0   8.6093 -2.7457  12.0805  7.8928  10.5825   
3            train_3       0  11.0604 -2.1518   8.9522  7.1957  12.5846   
4            train_4       0   9.8369 -1.4834  12.8746  6.6375  12.2772   
...              ...     ...      ...     ...      ...     ...      ...   
199995  train_199995       0  11.4880 -0.4956   8.2622  3.5142  10.3404   
199996  train_199996       0   4.9149 -2.4484  16.7052  6.6345   8.3096   
199997  train_199997       0  11.2232 -5.0518  10.5127  5.6456   9.3410   
199998  train_199998       0   9.7148 -8.6098  13.6104  5.7930  12.5173   
199999  train_199999       0  10.8762 -5.7105  12.1183  8.0328  11.5577   

          var_5   var_6    var_7  ...  var_190  var_191  var_192  var_193  \
0       -9.2834  5.1187  18.6266  ...   4.4354   3.9642   3.1364   1.6910   
1        7.0433  5.6208  16.5338  ...   7.6421   7.7214   2.5837  10.9516   
2       -9.0837  6.9427  14.6155  ...   2.9057   9.7905   1.6704   1.6858   
3       -1.8361  5.8428  14.9250  ...   4.4666   4.7433   0.7178   1.4214   
4        2.4486  5.9405  19.2514  ...  -1.4905   9.5214  -0.1508   9.1942   
...         ...     ...      ...  ...      ...      ...      ...      ...   
199995  11.6081  5.6709  15.1516  ...   6.1415  13.2305   3.9901   0.9388   
199996 -10.5628  5.8802  21.5940  ...   4.9611   4.6549   0.6998   1.8341   
199997  -5.4086  4.5555  21.5571  ...   4.0651   5.4414   3.1032   4.8793   
199998   0.5339  6.0479  17.0152  ...   2.6840   8.6587   2.7337  11.1178   
199999   0.3488  5.2839  15.2058  ...   8.9842   1.6893   0.1276   0.3766   

        var_194  var_195  var_196  var_197  var_198  var_199  
0       18.5227  -2.3978   7.8784   8.5635  12.7803  -1.0914  
1       15.4305   2.0339   8.1267   8.7889  18.3560   1.9518  
2       21.6042   3.1417  -6.5213   8.2675  14.7222   0.3965  
3       23.0347  -1.2706  -2.9275  10.2922  17.9697  -8.9996  
4       13.2876  -1.5121   3.9267   9.5031  17.9974  -8.8104  
...         ...      ...      ...      ...      ...      ...  
199995  18.0249  -1.7939   2.1661   8.5326  16.6660 -17.8661  
199996  22.2717   1.7337  -2.1651   6.7419  15.9054   0.3388  
199997  23.5311  -1.5736   1.2832   8.7155  13.8329   4.1995  
199998  20.4158  -0.0786   6.7980  10.0342  15.5289 -13.9001  
199999  15.2101  -2.4907  -2.2342   8.1857  12.1284   0.1385  

[200000 rows x 202 columns]

# 𝑃(𝑡𝑎𝑟𝑔𝑒𝑡=0|(1≤𝑣𝑎𝑟0≤15)∩(−10≤𝑣𝑎𝑟1≤9.5)∩(4≤𝑣𝑎𝑟2≤18))

# Que 1

In [6]:
boolean_mask1 = (df['target'] == 0)

In [7]:
df[boolean_mask1].shape[0]

179902

In [8]:
boolean_mask = (1<=df['var_0']) & (df['var_0'] <=15) & (-10 <= df['var_1']) & (df['var_1'] <= 9.5 ) &(4<=df['var_2']) & (df['var_2'] <=18)

In [9]:
df[boolean_mask].shape[0]

181161

In [10]:
frequency = df[boolean_mask].shape[0]

In [11]:
joint_probability = df[boolean_mask].shape[0] / df.shape[0]

In [12]:
joint_probability

0.905805

In [13]:
boolean_mask1 =  ((1<=df['var_0']) & (df['var_0'] <=15)) & ((-10 <= df['var_1']) & (df['var_1'] <= 9.5 )) & ( (4<=df['var_2']) & (df['var_2'] <=18)) & (df['target']== 0)

In [14]:
df[boolean_mask1].shape[0]

163891

In [15]:
combine_probability = df[boolean_mask1].shape[0]/df.shape[0]

In [16]:
combine_probability

0.819455

In [17]:
conditional_probability_que1 = combine_probability/joint_probability

In [18]:
conditional_probability_que1

0.9046704312738394

# 𝑃((1≤𝑣𝑎𝑟0≤15)∩(−10≤𝑣𝑎𝑟1≤9.5)∩(4≤𝑣𝑎𝑟2≤18)|𝑡𝑎𝑟𝑔𝑒𝑡=0)

# Que 2

In [19]:
 boolean_mask1=  ((1<=df['var_0']) & (df['var_0'] <=15)) & ((-10 <= df['var_1']) & (df['var_1'] <= 9.5 )) & ( (4<=df['var_2']) & (df['var_2'] <=18)) & (df['target']== 0)

In [20]:
df[boolean_mask1].shape[0]

163891

In [21]:
combine_probability = df[boolean_mask1].shape[0]/df.shape[0]

In [22]:
combine_probability

0.819455

In [23]:
boolean_mask = (df['target'] == 0)

In [24]:
df[boolean_mask].shape[0]

179902

In [25]:
joint_probability = df[boolean_mask].shape[0] / df.shape[0]

In [26]:
conditional_probability_que2 = combine_probability/joint_probability

In [27]:
conditional_probability_que2

0.9110015452857667

# 𝑃(𝑡𝑎𝑟𝑔𝑒𝑡=1|(1≤𝑣𝑎𝑟0≤15)∩(−10≤𝑣𝑎𝑟1≤9.5)∩(4≤𝑣𝑎𝑟2≤18))

# Que 3

In [28]:
 boolean_mask1=  ((1<=df['var_0']) & (df['var_0'] <=15)) & ((-10 <= df['var_1']) & (df['var_1'] <= 9.5 )) & ( (4<=df['var_2']) & (df['var_2'] <=18)) & (df['target']== 1)

In [29]:
df[boolean_mask1].shape[0]

17270

In [30]:
combine_probability = df[boolean_mask1].shape[0]/df.shape[0]

In [31]:
df[boolean_mask].shape[0]

179902

In [32]:
joint_probability = df[boolean_mask].shape[0] / df.shape[0]

In [33]:
conditional_probability_que3 = combine_probability/joint_probability

In [34]:
conditional_probability_que3

0.0959967093195184

# 𝑃((1≤𝑣𝑎𝑟0≤15)∩(−10≤𝑣𝑎𝑟1≤9.5)∩(4≤𝑣𝑎𝑟2≤18)|𝑡𝑎𝑟𝑔𝑒𝑡=1)

# Que 3

In [35]:
boolean_mask = (df['target'] == 1)

In [36]:
boolean_mask

0         False
1         False
2         False
3         False
4         False
          ...  
199995    False
199996    False
199997    False
199998    False
199999    False
Name: target, Length: 200000, dtype: bool

In [37]:
df[boolean_mask].shape[0]

20098

In [38]:
df[boolean_mask1].shape[0]

17270

In [39]:
combine_probability = df[boolean_mask1].shape[0]/df.shape[0]

In [40]:
joint_probability = df[boolean_mask].shape[0] / df.shape[0]

In [41]:
conditional_probability_que4 = combine_probability/joint_probability

In [42]:
conditional_probability_que4

0.8592894815404518

# conditional_probability_que1 =  0.9046704312738394

# conditional_probability_que2 =0.9110015452857667

# conditional_probability_que3 = 0.0959967093195184

# conditional_probability_que4 = 0.8592894815404518